### ISSUE TITLE
ListAssert inherits from IterableAssert and remove AbstractIterableAssert ?

### ISSUE THREAD

**Issue by nfrancois** [alexruiz/fest-assert-2.x#38]
I just remember of this TODO.
For recall, ListAssert header is :

``` java
public class ListAssert<T> extends AbstractIterableAssert<ListAssert<T>, List<T>, T> implements IndexedObjectEnumerableAssert<T>
```

I think we should, there is not reason against and to have better logical hierarchy.

**Comment by ash2k** [comment]
No, this will break returned self type on methods from `IterableAssert` in `ListAssert` (we loose `S` generic when extend).

**Comment by ash2k** [comment]
I think we should move all the code from `ListAssert` to the new class `AbstractListAssert` and make `ListAssert` extend it. This will enable library users to extend `AbstractListAssert` (like other abstract assert classes) and take all assertions from it - this improves extensibility of lib.

Also, this "idea" is applicable to other assert implementations that do not have abstract classes. Maybe move all code from concrete classes to abstract ones (also respecting type hierarchy)?

**Comment by joel-costigliola** [comment]
@ash2k I tend to agree with you, let's start with `ListAssert` and `AbstractListAssert`, we will do the remaining classes later to avoid a too big workload.
